sled-agent/src/sim/storage.rs

```diff
@@ -1883,6 +1906,9 @@ impl StorageInner {
     ) -> Arc<CrucibleData> {
         match self.get_dataset(zpool_id, dataset_id) {
             DatasetContents::Crucible(crucible) => crucible.data.clone(),
+            DatasetContents::LocalStorage(_) => {
+                panic!("asked for Crucible, got LocalStorage!")
```

Comment:
Maybe this is fine since it's the simulator, but should `get_crucible_dataset()` return an `Option<Arc<CrucibleData>>` and let the caller unwrap should they so choose? (Same question about `get_local_storage_dataset()` below)